### ISSUE TITLE
Feature Request: Rerun test manually

### ISSUE THREAD

**Issue by greenmooseSE** [wallabyjs/public#2781]
#### Scenario:
I am troubleshooting a specific test (IntelliJ) and want to rerun it manually (e.g. to see different log outputs that are generated) without having to modify code.

#### Feature suggestions:
1. Be able to rerun a test via an icon in wallaby app
2. Be able to rerun a test which is currently "in focus" in IDE, via some keyboard shortcut.
3. Be able to rerun tests from anywhere in covered code by executing something similar to `rerun tests covering line`, which should respect test filtering (e.g. only rerun 1 test if I have filtered on that test).

**Comment by ArtemGovorov** [comment]
>  I would want a 'run tests manually' kind of thing where I can change code but keep wallaby from rerunning, and then only run the test I have focused.

Do I understand it correctly that it sounds like you want Wallaby to be off most of the time and running a single test occasionally? If it is the case, then a combination of starting Wallaby on a [particular test file](https://wallabyjs.com/docs/intro/selected-tests.html#exclusive-test-run) and then [focusing a specific test](https://wallabyjs.com/docs/intro/selected-tests.html#automatic-test-focusing) to make Wallaby only run that test is the best solution.

If I have misunderstood or misinterpreted your workflow, can you please provide some more details about it, like why exactly you don't Wallaby to run your tests continuously and want a manual trigger instead?

**Comment by greenmooseSE** [comment]
> If I have misunderstood or misinterpreted your workflow, can you please provide some more details about it, like why exactly you don't Wallaby to run your tests continuously and want a manual trigger instead?

Yeah so I am troubleshooting an integration test (I got red errors in console where wallaby states that sandbox is inactive but no tests are failing, `npm test` worked fine). I am then starting the binary exclusion/inclusion of code to find the issue (reason seemed to be to be circular import of a file).
The restApi that tests are executing towards is running locally and is outputting valuable debug information. Therefore I do not want wallaby get in my way "until I am ready to rerun the test", but there is no way of doing that unless I disable/re-enable wallaby but that will rerun all tests (right?) and then the rest endpoint debug output will be lost in the wall of text caused by wallaby test execution.

Furthermore, the console output from wallaby just keeps appending so my workflow becomes "alter code, within ~5s clear the wallaby console, wait for test to finish, examine console output to see if I fixed issue, if not, redo process". It would be nice if it had at least auto-scroll or (or in my scenario "auto-clear" after each test run) and timestamps.

One workaround is to disable wallaby and in my terminal instead run a specific test via `npm` but then off course I loose the point with using wallaby.

p.s. I am also a fan of NCrunch and it has a great 'engine mode' customization where 'run tests manually' is  one engine mode setting I often used when troubleshooting long running tests d.s.

**Comment by ArtemGovorov** [comment]
It sounds like `Make current session run only on save` should work best for your scenario. Your tests will run only when you save the file. So you may make any number of edits, then whenever ready, save the file and Wallaby will run the test that you have been editing. Coverage markers disappear in this mode by design, because once you start editing, the markers become obsolete because tests are not running.

> Furthermore, the console output (inside IntelliJ) from wallaby just keeps appending so my workflow becomes "alter code, within ~5s clear the wallaby console, wait for test to finish, examine console output to see if I fixed issue, if not, redo process". It would be nice if it had at least auto-scroll or (or in my scenario "auto-clear" after each test run) and timestamps.

Wallaby Console output is not meant to be monitored, what's being displayed there is mostly to troubleshoot Wallaby itself, rather than your tests. What kind of output are you examining there (that can not be examined in The Failing Tests or in a specific test output)?

> One workaround is to disable wallaby and in my terminal instead run a specific test via npm but then off course I loose the point with using wallaby.

The combination of starting Wallaby on a [particular test file](https://wallabyjs.com/docs/intro/selected-tests.html#exclusive-test-run) and then [focusing a specific test](https://wallabyjs.com/docs/intro/selected-tests.html#automatic-test-focusing) to make Wallaby only run that test allows you to do the same but with Wallaby.

**Comment by greenmooseSE** [comment]
> It sounds like Make current session run only on save should work best for your scenario

Yeah but rather it is the "least bad" option because I must save file in order for the restApi endpoint to restart, and then test might fail depending on the race condition and I have to rerun it,. Furthermore there are cases where I want to save without running test (e.g. using postman to test a specific scenario for troubleshooting restApi output).

>  What kind of output are you examining there (that can not be examined in The Failing Tests or in a specific test output)? 

The unresponsive sandbox errors in this case.

> The combination of starting Wallaby on a particular test file and then focusing a specific test to make Wallaby only run that test allows you to do the same but with Wallaby.

No since it always executes test on code modification (when saved).
